tol_lab_share/messages/properties/complex/scientific_name_from_taxon_id.py

```diff
@@ -31,28 +41,17 @@ def value(self) -> Any:
         """Resolves the value of the scientific name by querying the EBI service
         for the value given to the taxon id stored in current instance. It also
         caches the value so it can be reused by other objects.
-        It also performs a very simple cache size releasing to avoid memory leaking.
         """
-        if self._input.value not in CACHE_TAXON_IDS:
-            if len(CACHE_TAXON_IDS) > MAX_CACHE_SIZE:
-                ScientificNameFromTaxonId.reset_cache()
-
-            r = get(f"{ self.taxonomy_url }/{ self._input.value }")
-
-            self._success = r.status_code == codes.ok
-
-            if self._success:
-                json = r.json()
-                CACHE_TAXON_IDS[self._input.value] = json["scientificName"]
-            else:
-                self.trigger_error(
-                    error_codes.ERROR_14_PROBLEM_ACCESSING_TAXON_ID,
-                    text=f"HTTP code {r.status_code} - Problem when accessing taxon id {self._input.value}",
-                )
-
-        return CACHE_TAXON_IDS[self._input.value]
+        try:
+            return get_taxon_id_from_ebi(self.taxonomy_url, self._input.value)
+        except StatusCodeException as e:
+            self.trigger_error(
+                error_codes.ERROR_14_PROBLEM_ACCESSING_TAXON_ID,
+                text=f"HTTP code {e.status_code} - Problem when accessing taxon id {self._input.value}",
+            )
+            return None
 
     @staticmethod
     def reset_cache():
         """Static method to reset the cache shared by all instances"""
```

Comment:
This is still a static method though.  It's more of a convenience to be honest.  Someone could call the clear method directly if they wanted to.  I thought this made it clearer that the cache is associated with the usage of this class so I left it in.

Reply:
Maybe we're talking about different things.  The cache itself was being manually implemented (quite badly) here and I changed it to using the built-in `lru_cache` from Python's functools.  The new function will cache responses from the `get` requests up to 1000 of them, then it drops the least recently used one from the cache.

The method with this comment in, that you're referencing, is static though, as seen by the `@staticmethod` written above it.  This means you can call it without having an instance of the class, which makes sense since the cache function is outside this class and therefore shared across all instances.